cmd/trivy/main.go

```diff
@@ -1,15 +1,30 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"os"
+
 	"github.com/aquasecurity/trivy/pkg/commands"
 	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/plugin"
 )
 
 var (
 	version = "dev"
 )
 
 func main() {
+	// Trivy behaves as the specified plugin.
+	if runAsPlugin := os.Getenv("TRIVY_RUN_AS_PLUGIN"); runAsPlugin != "" {
+		if !plugin.IsPredefined(runAsPlugin) {
+			log.Fatal(fmt.Errorf("unknown plugin: %s", runAsPlugin))
+		}
+		if err := plugin.RunWithArgs(context.Background(), runAsPlugin, os.Args); err != nil {
+			log.Fatal(err)
+		}
```

Comment:
hi guys! sorry for interfering.

maybe should we interrupt `trivy` if `plugin.RunWithArgs` has no error?

Reply:
Oh, nice catch. Thanks.

Reply:
@afdesk Could you take a look again?
https://github.com/aquasecurity/trivy/pull/2577/commits/466cd3b4a9ea3840950802abe8ce1dc306223d72

Reply:
@knqyf263 LGTM

p.s. do i understand correctly that this option works only for official plugins (`trivy-plugin-kubectl` and `trivy-plugin-aqua`)?

Reply:
Yes, I want to restrict the use case.